src/object_based/extent.hpp

```diff
@@ -68,29 +94,52 @@ namespace ear {
     double operator()(const Eigen::Ref<Eigen::RowVector3d> &position) const;
 
    private:
+    // pointer to the actual weighting function, one of the static weight_*
+    // below
+    double (*weight_cb)(const WeightingFunction &self,
+                        const Eigen::Ref<Eigen::RowVector3d> &position);
+
+    static double weight_circle(const WeightingFunction &self,
+                                const Eigen::Ref<Eigen::RowVector3d> &position);
+    static double weight_stadium(
+        const WeightingFunction &self,
+        const Eigen::Ref<Eigen::RowVector3d> &position);
+
     const double _fadeWidth = 10.0;
     double _width;
     double _height;
-    double _circleRadius;
+    // angle of the circle centres from the source position
+    double circlePos() const { return _width - _height; }
+    double circleRadius() const { return _height; }
     Eigen::Matrix3d _flippedBasis;
-    double _circlePos;
-    Eigen::Matrix<double, 3, 2> _circlePositions;
+
+    // xy position of the right circle centre, used to measure the distance via
+    // dot product
+    Eigen::Vector2d right_circle_centre;
```

Comment:
should these have a leading underscore for consistency?

Reply:
Yeah, I should probably go through and change everything to underscored camelCase. I just really don't like this style and would rather move away from it, so I might just make the existing files consistent but leave the new ones as is -- there's already a fair chunk of lowercase_underscore code around.